charts/skypilot/templates/api-deployment.yaml

```diff
@@ -61,12 +63,24 @@ spec:
           {{- if .Values.apiService.preDeployHook }}
           {{ .Values.apiService.preDeployHook | nindent 10 }}
           {{- end }}
-          {{- if .Values.apiService.config }}
           mkdir -p /root/.sky
-          echo "Linking config.yaml from ConfigMap \`skypilot-config\` to /root/.sky/config.yaml"
-          # The configmap serves as the ground truth for the config.yaml file, read-only
-          ln -sf /var/skypilot/config/config.yaml /root/.sky/config.yaml
-          {{- end }}
+          # When the config.yaml is a symlink, it should be from the old API
+          # server code. We remove the symlink and copy the ConfigMap config to
+          # PVC location for backward compatibility.
+          # TODO(zhwu): remove this after 0.12.0.
+          if [ -L /root/.sky/config.yaml ]; then
+            echo "Config.yaml is a symlink to ConfigMap config, deleting symlink"
+            rm /root/.sky/config.yaml
+          fi
```

Comment:
Without symlink, we lost the capability to update config maps with `helm upgrade` without restarting the API server. Maybe we need some followup fix for this?

Reply:
Yes, we mentioned it here: https://docs.skypilot.co/en/editable-workspace/reference/api-server/api-server-admin-deploy.html#optional-update-skypilot-configuration

We need to figure out a way to fix this. A potential way is to expose an API for SkyPilot config get and update, and only allow user to update the SkyPilot config with SkyPilot's own config API instead of relying on configMap